.travis.yml

```diff
@@ -1,50 +1,41 @@
-dist: focal
-
-group: stable
-
 language: python
 
-cache: pip
+dist: jammy
 
-notifications:
-  email: true
-
-matrix:
-  include:
-    - python: 3.8
-    - python: 3.9
-    - python: 3.10
-    - python: 3.11
+stages:
+  - name: Build-Test
+    if: tag IS blank
+ - name: Semantic-Release
+   if: (branch = main) AND (type IN (push, api)) AND (fork = false)
+ - name: Publish-Release
+   if: (tag IS present) AND (fork = false)
 
-install:
-- sudo apt-get update
-- sudo apt-get install pandoc
-- pip install pypandoc
-- echo -e "machine github.ibm.com\n  login $GITHUB_ACCESS_TOKEN" > ~/.netrc
-- pip install setuptools=="60.8.2"
+install: true
 
 script:
 - make ci
 
-before_deploy:
-  - nvm install 18
-  - node --version
-  - npm --version
-  - npm install
-  - pip install bump-my-version
-
-deploy:
-  - provider: script
-    script: npm run semantic-release
-    skip_cleanup: true
-    on:
-      python: '3.8'
-      branch: main
-  - provider: pypi
-    user: __token__
-    password: $PYPI_TOKEN
-    repository: https://upload.pypi.org/legacy
-    skip_cleanup: true
-    on:
+jobs:
+  include:
+    - stage: Build-Test
       python: '3.8'
-      tags: true
\ No newline at end of file
+    - python: '3.10'
+    - python: '3.11'
+    - python: '3.12'
+
+   - stage: Semantic-Release
+     language: node_js
+     node_js: 22
+     install:
+       - npm install
+       - pip install --user bump-my-version
+     script:
+       - npm run semantic-release
```

Comment:
make sure your package.json contains a "semantic-release" script definition.  this would be easy to miss when copying the changes.